### PR TITLE
[android][ios] Upgrade @react-native-picker/picker to 2.2.1

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Fixes [ENG-2541](https://linear.app/expo/issue/ENG-2541/react-native-pickerpicker-210-221)

# How

- `$ et uvm --module "@react-native-picker/picker" --commit "v2.2.1" --target "expo-go"`
- `$ yarn && et pod-install -f`

# Test Plan

- [X] Android (unversioned) works
- [x] iOS (unversioned) works

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/bare-expo/ios/Podfile.lock [hunk 1]
```diff
@@ -633,7 +633,7 @@ PODS:
     - React
   - react-native-viewpager (5.0.11):
     - React-Core
-  - react-native-webview (11.13.0):
```
Comment (byCedric) [inline apps/bare-expo/ios/Podfile.lock:636]: Not sure why webview is being updated, will check tomorrow!

Comment (brentvatne) [reply apps/bare-expo/ios/Podfile.lock:636]: it looks like we didn't run pod install in bare-expo after merging #15330

**Comment by tsapeta** [comment]
I've rerun failing iOS and Android Client workflows as they were caused by network issues. Home app seems to be failing due to different reasons 🤔

**Comment by byCedric** [comment]
Not sure why, but the new `@react-native-picker/picker` file references were messed up. They were referring `.../Api/Components/RNC...` instead of `.../Api/Components/Picker/RNC...`. Plus some were duplicated (probably bc of the wrong path).

**Comment by brentvatne** [comment]
looks like ios build is failing again, seems like possibly a source file is missing

**Comment by byCedric** [comment]
> looks like ios build is failing again, seems like possibly a source file is missing

I'm not exactly sure where this is coming from. Nuking my repo, checking out this branch, and build it for my phone works fine. I can take a look at it tomorrow though, but might be better to get this merged asap?

**Comment by byCedric** [comment]
Ok, this should do it. I can build and use it locally (unversioned) and the new picker works fine. 🤞  CI succeeds, and if so feel free to merge it. @brentvatne

**Comment by byCedric** [comment]
Ok so here is what I found. I retried to vendor this library [on branch `@bycedric/picker/revendor-with-fixed-ios-grouping`](https://github.com/expo/expo/compare/master...@bycedric/picker/revendor-with-fixed-ios-grouping).

- Somehow `@react-native-picker/picker` was previously set without a group. Instead of a group like `Maps`, and a folder prefix of `maps` on the pbxproj group, each individual picker file had their path prefixed by `Picker/...`.
- It seems like using individual files with prefixed path doesn't play nice with `et uvm`, and should only be prefixed by group instead.
- Even after manually correcting this, the build will still fail on the error:
    > Build input file cannot be found: '/Users/cedric/Projects/expo/expo/ios/Exponent/Supporting/EXBuildConstants.plist'

I tried this from the normal local state of the repo, and I tried it from a fully nuked and cleaned repo. Seems like it might not be related to this PR though. Anyone knows the best way forward?

It's kind of remarkable that even without the manually fixed pbxproj groups (on this branch), it seems to build fine locally (and the picker works fine too)

**Comment by byCedric** [comment]
ok, so let's go for this PR instead: #15400, it does seem to build but the test suite failed on iOS due to some detox issue I guess. (@brentvatne)

**Pull request closed without merging**